diag_manager/diag_util.F90

```diff
@@ -2499,6 +2499,86 @@ SUBROUTINE prepend_attribute_file(out_file, att_name, prepend_value, err_msg)
     END IF
   END SUBROUTINE prepend_attribute_file
 
+  !> @brief Updates indices based on presence/absence of input indices is, js, ks, ie, je, and ke.
+  ! Computes halo sizes in x and y directions.
+  ! This routine is intended to be used in diag manager.
+  !> @return .false. if there is no error else .true.
+  function recondition_indices(indices, data, is_in, js_in, ks_in, &
+    ie_in, je_in, ke_in, err_msg) result(ierr)
+    integer, intent(inout) :: indices(12) !< Stores indices in order:
+                                              !! (/is, js, ks, ie, je, ke, hi, fis, fie, hj, fjs, fje/)
+    class(*), intent(in) :: data(:,:,:,:) !< Dummy variable; only the sizes of the first 3 dimensions are used
+    integer, intent(in), optional :: is_in, js_in, ks_in, ie_in, je_in, ke_in !< User input indices
+    character(len=*), intent(out), optional :: err_msg !< Error message to pass back to caller
+    logical :: ierr !< Error flag
+
+    integer :: is, js, ks, ie, je, ke !< Local indices to update
+    integer   :: hi !< halo size in x direction
+    integer   :: hj !< halo size in y direction
```

Comment:
These in the I and J dimension not the X and Y direction

Reply:
done